src/bootstrap/bootstrap.py

```diff
@@ -465,16 +470,20 @@ def downloading_llvm(self):
 
     def _download_stage0_helper(self, filename, pattern, tarball_suffix, date=None):
         if date is None:
-            date = self.date
+            date = self.stage1_commit or self.date
         cache_dst = os.path.join(self.build_dir, "cache")
         rustc_cache = os.path.join(cache_dst, date)
         if not os.path.exists(rustc_cache):
             os.makedirs(rustc_cache)
 
-        url = "{}/dist/{}".format(self._download_url, date)
+        if self.stage1_commit is not None:
+            url = "https://ci-artifacts.rust-lang.org/rustc-builds/{}".format(self.stage1_commit)
```

Comment:
We should continue to use the stage0.txt rustfmt so I would not try to do this here.

Reply:
Hmm, that makes sense, but I need to change the URL for downloading commit artifacts instead of nightlies. I moved the logic out to `download_stage0()` instead.